File: src/main/java/me/nepnep/nepbot/JoinMessage.java
package me.nepnep.nepbot;

import me.nepnep.nepbot.database.JoinMessageDatabase;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.TextChannel;
import net.dv8tion.jda.api.events.guild.member.GuildMemberJoinEvent;
import net.dv8tion.jda.api.exceptions.InsufficientPermissionException;
import net.dv8tion.jda.api.hooks.ListenerAdapter;

import java.util.Map;

public class JoinMessage extends ListenerAdapter {
    @Override
    public void onGuildMemberJoin(GuildMemberJoinEvent event) {
        try {
            Guild guild = event.getGuild();

            Map<String, String> map = JoinMessageDatabase.getJoinDetails(guild.getIdLong());
            if (map != null) {
                String channelId = map.get("channel");

                TextChannel channel = guild.getTextChannelById(channelId);
                if (channel != null) {
                    String message = map.get("message");
                    channel.sendMessage(message.replace("%s", event.getUser().getAsMention())).queue();
                }
            }
        } catch (InsufficientPermissionException e) {
            // Intentionally empty
        }
    }
}

File: src/main/kotlin/me/nepnep/nepbot/JoinMessage.kt
package me.nepnep.nepbot

import me.nepnep.nepbot.database.getJoinDetails
import net.dv8tion.jda.api.events.guild.member.GuildMemberJoinEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter

class JoinMessage : ListenerAdapter() {
    override fun onGuildMemberJoin(event: GuildMemberJoinEvent) {
        val guild = event.guild
        val joinDetails = guild.getJoinDetails() ?: return
        val channel = guild.getTextChannelById(joinDetails["channel"].longValue()) ?: return
        val message = joinDetails["message"].textValue()

        channel.sendMessage(message.replace("%s", event.user.asMention)).queue()
    }
}